backend/modules/publishing/service.py

```
@@ -1 +1,603 @@
+        # Check if metadata already has S3 URL
+        existing_url = None
+        if wasabi_url and _is_s3_url(wasabi_url):
+            existing_url = wasabi_url
+        elif pdf_url and _is_s3_url(pdf_url):
+            existing_url = pdf_url
+        
+        if existing_url:
+            self.repository.create_or_update_scenario_file(
+                simulation.id, filename, existing_url, file_size, file_type
+            )
+            setattr(simulation, 'case_study_url', existing_url)
```

Ruff (0.14.7)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)